### PR TITLE
Insights troubleshooting

### PR DESCRIPTION
**Pull request by strika**

### CODE REVIEW

**Comment by strika** [comment]
@dunjardl Please review.

File: source/docs/insights-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: Insights troubleshooting
+category: Insights troubleshooting
+---
+
+If you turned on Insights for a project on Semaphore, but you don't see any test
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: If you turned on Insights -> If you have turned Insights on

File: source/docs/insights-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+files on the Insights page or some stats are missing, this guide will help you
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: page, or (add comma)

File: source/docs/insights-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+find the source of the problem.
+
+Semaphore supports insights for RSpec and Cucumber testing frameworks. If you
+use some other framework, but you would still like to use Insights, please [let
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: use -> are using

File: source/docs/insights-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+layout: post
+title: Insights troubleshooting
+category: Insights troubleshooting
+---
+
+If you turned on Insights for a project on Semaphore, but you don't see any test
+files on the Insights page or some stats are missing, this guide will help you
+find the source of the problem.
+
+Semaphore supports insights for RSpec and Cucumber testing frameworks. If you
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: insights -> Insights (capitalisation)

File: source/docs/insights-troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+use some other framework, but you would still like to use Insights, please [let
+us know](https://renderedtext.typeform.com/to/d1fWsl) by answering few short
+questions.
+
+## RSpec
+
+At the moment, Semaphore supports only RSpec 3.x. If you use earlier versions of
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: use -> are using

File: source/docs/insights-troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+RSpec, it's likely that you will see test files on the Insights page, but you
+won't see all stats. Percentage of passed builds will be visible, but stats for
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: won't see all stats. Percentage -> will not be able to see all the stats. The percentage
but stats -> but the stats

File: source/docs/insights-troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+mean time, max time and standard deviation will be missing. The solution is to
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: The solution is to -> The solution to this problem is to

File: source/docs/insights-troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+upgrade to a newer version of RSpec.
+
+If you use RSpec 3.x, but you don't see any test files on the Insights page,
+there are few possible causes described below.
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: there are few possible causes -> there are a few

File: source/docs/insights-troubleshooting.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+#### All test files finish faster than 10 seconds
+
+Semaphore Insights are displayed only for test files that take more than 10
+seconds to finish.
+
+#### Your options are defined with SPEC_OPTS environment variable
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: defined with SPEC_OPTS environment variable 0> defined with the SPEC_OPTS environment variable

File: source/docs/insights-troubleshooting.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+There are few ways to pass configuration options to RSpec runner:
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: There are few ways -> There are a few ways

File: source/docs/insights-troubleshooting.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+- `SPEC_OPTS` environment variable
+- `.rspec` file
+- command line options
+
+When you turn on RSpec Insights on a project, Semaphore will generate `.rspec`
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: When you turn on RSpec Insights on a project -> When you turn RSpec Insights on for a project

File: source/docs/insights-troubleshooting.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+file, if it's not already present and inject `--format json` option in the file.
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: file, if it's not already present and inject `--format json` option -> file if it does not already exist, and inject the `--format json` option

File: source/docs/insights-troubleshooting.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+However, if your project is using `SPEC_OPTS` environment variable to pass
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: using `SPEC_OPTS` environment variable -> using the `SPEC_OPTS` environment variable

File: source/docs/insights-troubleshooting.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+options to the RSpec runner, they will override the options in `.rspec` file and
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: file, and

File: source/docs/insights-troubleshooting.md [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+Semaphore will not be able to gather stats for the Insights.
+
+A typical RSpec command with `SPEC_OPTS` might look like this:
+
+```bash
+bundle exec rake spec SPEC_OPTS="--format documentation"
+```
+
+To use Insights, replace `SPEC_OPTS` environment variable with `.rspec` file.
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: replace `SPEC_OPTS` environment variable with `.rspec` file -> replace the `SPEC_OPTS` environment variable with an `.rspec` file

File: source/docs/insights-troubleshooting.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+You can commit the file to the repository. In that case, options will be used on
```
Comment (strika) [inline source/docs/insights-troubleshooting.md]: In that case, these options...

File: source/docs/insights-troubleshooting.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+all development machines, unless a developer overrides them. If you don't want
+to commit `.rspec` file to the project repository, you can create the file using
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: commit `.rspec` file -> commit the `.rspec` file

File: source/docs/insights-troubleshooting.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+[custom configuration files](/docs/adding-custom-configuration-files.html). That
+way, `.rspec` file will be available only on Semaphore.
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: way, `.rspec` -> way, the `.rspec`

File: source/docs/insights-troubleshooting.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+
+#### Your tests are executed using parallel_test gem
+
+If you run your RSpec tests using parallel_test gem the report with stats for
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: gem, the report

File: source/docs/insights-troubleshooting.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+each test example will not be in valid format and you will not be able to use
```
Comment (dunjardl) [inline source/docs/insights-troubleshooting.md]: format, and

**Pull request merged**